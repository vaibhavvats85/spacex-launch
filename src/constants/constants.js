export const APP_TITLE = 'SpaceX Launch Programs';
export const DEVELOPER_NAME = 'Vaibhav Vats';
export const DEVELOPED_BY = 'Developed by';
export const MISSION_IDS = 'Mission Ids';
export const LAUNCH_YEAR = 'Launch Year';
export const SUCCESSFULL_LAUNCH = 'Successful Launch';
export const SUCCESSFULL_LANDING = 'Successful Landing';
export const FILTERS = 'Filters';
export const LAUNCH_ERROR = 'Unable to Load Launch Programs. We will get bakc to you soon.';
export const NO_PROGRAM = 'No programs available for the given Criteria';
export const LAUNCH_YEAR_KEY = 'launch_year';
export const LAUNCH_SUCCESS_KEY = 'launch_success';
export const LAND_SUCCESS_KEY = 'land_success';
