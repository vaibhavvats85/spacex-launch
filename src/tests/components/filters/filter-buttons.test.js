import { shallow } from 'enzyme';
import * as constants from '../../../constants';
import React from 'react';
import FilterButtons from '../../../components/filters/filter-buttons';

let wrapper;
let props = {
    btns: ['2006', '2007'],
    type: constants.LAUNCH_YEAR_KEY,
    params: { get: jest.fn().mockReturnValue('2006') },
    clickEvent: jest.fn()
}
jest.mock('react-router-dom', () => ({
    useLocation: () => ({ search: '?launch_year=2006' }),
}));

beforeEach(() => {
    wrapper = shallow(<FilterButtons {...props} />);
});

test('should render FilterButtons component correctly', () => {
    expect(wrapper).toMatchSnapshot();
});


test('should check button correctly', () => {
    const active = wrapper.find('Button').at(0).prop('active').toLowerCase();
    const id = wrapper.find('Button').at(0).prop('value').id.toLowerCase();
    expect(id).toEqual(active)
});


test('should call handleClick event', () => {
    wrapper.find('Button').at(0).prop('handleClick')();
    expect(props.clickEvent).toHaveBeenCalled();
});
