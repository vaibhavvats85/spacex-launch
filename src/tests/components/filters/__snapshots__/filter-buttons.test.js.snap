// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FilterButtons component correctly 1`] = `
Array [
  <Button
    active="2006launch_year"
    group="launch_year"
    handleClick={[Function]}
    key="0"
    value={
      Object {
        "id": "2006launch_year",
        "val": "2006",
      }
    }
  />,
  <Button
    active="2006launch_year"
    group="launch_year"
    handleClick={[Function]}
    key="1"
    value={
      Object {
        "id": "2007launch_year",
        "val": "2007",
      }
    }
  />,
]
`;
